### PR TITLE
Remove legacy Meta, bring Optics KSP

### PR DESCRIPTION
**Pull request by nomisRev**
(Disclaimer: edited by @serras)

This PR cleans the slate for the Meta story:
- Completely removes Meta from the repo, everything is now in https://github.com/arrow-kt/arrow-meta.
- The Optics plug-in based on KSP now lives in **this** repo, since it does not use Meta anymore. As a result, a few functions from the `arrow-meta-test` package have been copied there.
- Updates the Kotlin version to 1.6.10 (the latest at this moment), this is required by both the `ksp` and `kotlin-compile-testing` packages.

The update of `kotlinx-coroutines` was tried, but it has been unsuccessful due to breaking changes in version 1.6. This would need further investigation.

### CODE REVIEW

**Comment by serras** [comment]
Should we move the Optics KSP module here? It no longer depends on Meta, but the docs here do depend on the plug-in.

**Comment by nomisRev** [comment]
We could move it into the Arrow Optics module yes, it will probably make the most sense to develop the compiler plugin there that. I hope it goes green with the bumped versions 🤞

File: settings.gradle.kts
Comment (raulraja) [inline settings.gradle.kts:35]: 🥳

**Review by raulraja — APPROVED** [review]

File: build.gradle.kts
Comment (nomisRev) [inline build.gradle.kts:72]: 👌

**Comment by nomisRev** [comment]
@serras IT'S GREEEN!!! 😍 😍 😍

**Comment by serras** [comment]
Finally! I’ll approve it, but I think you and @raulraja should have a look too and merge it if you are OK.

**Review by serras — APPROVED** [review]

File: arrow-libs/core/arrow-core/build.gradle.kts
Comment (nomisRev) [inline arrow-libs/core/arrow-core/build.gradle.kts:17]: @JavierSegoviaCordoba do you know if this is correct.
Since we expose types from Kotlin Std from public API, it should be marked as `api` right?

File: arrow-libs/optics/arrow-optics-test/src/commonTest/kotlin/arrow/optics/DSLTest.kt
Comment (nomisRev) [inline arrow-libs/optics/arrow-optics-test/src/commonTest/kotlin/arrow/optics/DSLTest.kt:1]: This file can be deleted, it's duplicated with `DSLTest.kt` inside `arrow-optics/test`

**Review by nomisRev — COMMENTED** [review]
Couple of small left overs we need to clean up.
Thank you for looking into this @serras! This is great 👏 👏 👏 🥳

**Comment by nomisRev** [comment]
@serras I cannot approve since I originally opened this PR. If we clean up these last leftovers it's ready to merge.

File: arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/EitherTest.kt
Comment (serras) [reply arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/EitherTest.kt:39]: Unfortunately not :(

**Comment by serras** [comment]
There seems to be a flaky test in `arrow-fx-stm`

**Comment by nomisRev** [comment]
@serras  mmm,  I saw `TMapTest` failed for a property that has multiple conflicting keys 🤔
Don't think we should block this PR for it, since it should be unrelated. I retriggered CI.

**Pull request merged**
